src/action.ts

```diff
@@ -0,0 +1,36 @@
+import { Step, SwapAction } from "./cnd/swaps_payload";
+import { Swap } from "./swap";
+import { Transaction } from "./transaction";
+
+/**
+ * An executable action.
+ */
+export class Action {
+  constructor(private action: SwapAction, private swap: Swap) {}
+
+  get name(): Step {
+    return this.action.name;
+  }
+
+  /**
+   * Execute the action, will fail if the action is not available anymore.
+   *
+   * @throws A {@link Problem} from the cnd REST API, or {@link WalletError} if the blockchain wallet throws, or an {@link Error}.
+   */
+  public async execute(): Promise<Transaction | string> {
+    const recommendedAction = await this.swap.nextAction();
+
+    if (!recommendedAction || this.name !== recommendedAction.name) {
+      throw new Error(
+        `Action ${
+          this.name
+        } is not recommended anymore. Current recommended action is: ${
+          recommendedAction ? recommendedAction.name : recommendedAction
+        }`
+      );
+    }
+
```

Comment:
I don't think we need this check. `cnd` MUST fail anyway if you try to do an action that no longer works.

You don't get any guarantees out of this because theoretically, the action you are holding on to could get stale between your check above here and where you execute it below :)

Reply:
Ah yes indeed.